src/transformers/pipelines/text_to_audio.py

```diff
@@ -241,29 +276,28 @@ def _sanitize_parameters(
         return preprocess_params, params, postprocess_params
 
     def postprocess(self, audio):
-        output_dict = {}
 
-        if self.model.config.model_type == "csm":
-            waveform_key = "audio"
-        else:
-            waveform_key = "waveform"
-
-        # We directly get the waveform
-        if self.no_processor:
-            if isinstance(audio, dict):
-                waveform = audio[waveform_key]
-            elif isinstance(audio, tuple):
-                waveform = audio[0]
+        needs_decoding = False
+        if isinstance(audio, dict):
+            if "audio" in audio:
+                audio = audio["audio"]
             else:
-                waveform = audio
-        # Or we need to postprocess to get the waveform
-        else:
-            waveform = self.processor.decode(audio)
-
-        if isinstance(audio, list):
-            output_dict["audio"] = [el.to(device="cpu", dtype=torch.float).numpy() for el in waveform]
+                needs_decoding = True
+                audio = audio["sequences"]
```

Comment:
@vasqu no more exceptions in postprocessing with such logic and by ensuring a dict when generate is used within `_forward`